src/otoole/write_strategies.py

```diff
@@ -81,11 +101,13 @@ def _write_parameter(
         if len(name) > 31:
             raise OtooleExcelNameLengthError(name=name)
 
-        df = self._form_parameter(df, parameter_name, default)
         if not df.empty:
+            df = self._form_parameter(df, parameter_name, default)
             df.to_excel(handle, sheet_name=name, merge_cells=False, index=True)
         else:
-            logger.info(f"Skipped writing {parameter_name} as it is empty")
+            logger.debug(f"Dataframe {parameter_name} is empty")
+            df = self._form_parameter_template(parameter_name, **kwargs)
+            df.to_excel(handle, sheet_name=name, merge_cells=False, index=False)
```

Comment:
Example of question 3, where I moved the check for for empty dataframes outside of the `_form_parameters(...)` method.